src/dspbase.jl

```diff
@@ -718,9 +718,14 @@ function conv(u::AbstractArray{<:BLAS.BlasFloat, N},
     conv(u, float(v))
 end
 
-function conv(A::AbstractArray, B::AbstractArray)
-    maxnd = max(ndims(A), ndims(B))
-    return conv(cat(A, dims=maxnd), cat(B, dims=maxnd))
+function conv(A::AbstractArray{<:Number, M},
+              B::AbstractArray{<:Number, N}) where {M, N}
+    if (M < N)
+        conv(cat(A, dims=N), B)
```

Comment:
While at it: `cat` is not type-stable, but I think we can at least make the return type of `conv` inferable by adding a type assertion like
```suggestion
        conv(cat(A, dims=N)::AbstractArray{typof(A), N}, B)
```
and similar below.

Reply:
Do you mean `eltype` instead of `typeof`?

Reply:
Oh yes, of course.